docs/doxygen/porting.txt

```diff
@@ -0,0 +1,77 @@
+/**
+@page mqtt_porting Porting Guide
+@brief Guide for porting MQTT to a new platform.
+
+A port to a new system must provide the following components:
+1. [Configuration Macros](@ref porting_config)
+2. [Transport Interface](@ref porting_transport)
+3. [Time Function](@ref porting_time)
+
+@section porting_config Configuration Macros
+@brief Settings that must be set as macros in the config header `core_mqtt_config.h`, or passed in as compiler options
```

Comment:
Providing the custom config is optional now. This should be updated to mention that either the custom config is required OR `MQTT_DO_NOT_USE_CUSTOM_CONFIG` macro can be passed as compiler option